### PR TITLE
Add disjunction / fallback tag expression

### PR DESCRIPTION
**Pull request by afontenot**
Check-list
----------

 * [x] There is a linked issue discussing the motivations for this feature or bugfix
 * [x] Unit tests have been added where possible
 * [x] I've added / updated documentation for any user-facing features.
 * [x] Performance seems to be comparable or better than current `main`

What this change is adding / fixing
-----------------------------------

Add a new piece of tag pattern syntax, with the operator `||`. This operator returns the tag pattern on the left side, if it is not empty, and otherwise returns the tag pattern on the right side.

It is implemented so as to be chainable, e.g. one can do

    <<composer>||<albumartist>||<artist>>

And the first non-empty expression will be used.

In documentation, this syntax will be called the "fallback" tag expression, for greater clarity.

Fixes https://github.com/quodlibet/quodlibet/issues/4152.

This PR also documents the new feature and improves the documentation for tag patterns to make it more accessible to new users. Tests have been added.

### CODE REVIEW

File: docs/guide/tags/patterns.rst [hunk 1]
```diff
@@ -3,79 +3,154 @@
...
+in more complex ways.
+
+The file renaming tool uses tag patterns to determine the file names and folder
+structure to be used for the moved files.
+
+Usage
+-----
+
+A tag pattern is a string (piece of text) in which certain symbols receive
+special meaning. Parts of the string that carry no special meaning will be
```
Comment (declension) [inline docs/guide/tags/patterns.rst:20]:
```suggestion
special meaning.
Parts of the string that carry no special meaning will be
```

(Generally I'm always trying now to edge towards [sembr](https://sembr.org/) for maintainability.)

**Review by declension — COMMENTED** [review]
This is great, thanks!

If you can be bothered to reflow markup with sembr then great, otherwise I'll do this some time later.

Like the new naming too.

**Comment by afontenot** [comment]
I haven't encountered sembr before, but I think I've formatted the document as requested.

**Comment by declension** [comment]
Thanks!

**Pull request merged**